packages/fe-fpm-writer/src/common/file.ts

```diff
@@ -0,0 +1,98 @@
+import type { Editor } from 'mem-fs-editor';
+import type { TabInfo } from '../common/types';
+
+const CHAR_SPACE = ' ';
+const CHAR_TAB = '\t';
+
+type WriteJsonReplacer = ((key: string, value: any) => any) | Array<string | number>;
+
+type WriteJsonSpace = number | string;
+interface ExtendJsonParams {
+    filepath: string;
+    content: string;
+    replacer?: WriteJsonReplacer;
+    tabInfo?: TabInfo;
+}
+
+/**
+ * Method returns tab info for passed line.
+ *
+ * @param line - line with tab spacing
+ * @returns tab size information
+ */
+function getLineTabInfo(line: string): TabInfo | undefined {
+    let tabSize: TabInfo | undefined;
+    const symbol = line[0] === CHAR_TAB ? CHAR_TAB : CHAR_SPACE;
+    // get count of tabs
+    for (let i = 0; i < line.length; i++) {
+        const char = line[i];
+        if (char !== symbol) {
+            tabSize = {
+                size: i,
+                useTabSymbol: symbol === CHAR_TAB
+            };
+            break;
+        }
+    }
+    return tabSize;
+}
+
+/**
+ * Method calculates tab space info for passed file content.
+ *
+ * @param content - file content.
+ * @returns tab size information.
+ */
+export function detectTabSpacing(content: string): TabInfo | undefined {
+    let tabSize: TabInfo | undefined;
+    const tabSymbols = [CHAR_SPACE, CHAR_TAB];
+    const lines = content.split(/\r\n|\n/);
+    const lineWithSpacing = lines.find((line: string): boolean => {
+        return tabSymbols.includes(line[0]);
+    });
+    if (lineWithSpacing) {
+        tabSize = getLineTabInfo(lineWithSpacing);
+    }
+    return tabSize;
+}
+
+/**
+ * Method calculates tab spacing parameter for 'JSON.stringify' method.
+ *
+ * @param fs - the mem-fs editor instance.
+ * @param filePath - path to file to read.
+ * @param tabInfo - External tab configuration.
+ * @returns tab size information.
+ */
+export function getJsonSpace(fs: Editor, filePath: string, tabInfo?: TabInfo | undefined): WriteJsonSpace | undefined {
+    let space: WriteJsonSpace | undefined;
+    // Use passed tab info or read from existing content
+    if (!tabInfo) {
+        const content = fs.read(filePath);
+        tabInfo = detectTabSpacing(content);
+    }
+    if (tabInfo !== undefined) {
```

Comment:
`else` would be sufficient

Reply:
@stefanschreck 

I applied some correction in code, but `else` will break logic - result of first if block can affect second if condition. Idea is:
1. 1st condition. if `tabInfo` is not passed - we calculate value for `tabInfo` based on passed file;
2. 2nd condition. It checks if `tabInfo` is defined and calculates value for `space` variable;

We can pass `tabInfo` as custom tab info configuration, but it is optional param. If it is not passed, then first 'if' calculates `tabInfo` by checking target file.

If we add `else`, then calculated value in 1st condition will never be used in second condition, but it should be:
![image](https://user-images.githubusercontent.com/90789422/210054431-cdbaad3f-d026-4712-8918-3bc75ba6ab5b.png)

I also added some additional tests with new commit to catch such case - https://github.com/SAP/open-ux-tools/pull/857/commits/b234f10d48967c9e5f84bffb7298ffa77c995c00

Please rereview